src/utils.js

```diff
@@ -204,11 +208,37 @@ export const parseQRCode = (data) => {
  * '<address>' -> '<address>'
  *
  * @param {string} plainText containing wallet address.
+ *
+ * @typedef ExtractedAddress
```

Comment:
From my experience, static code analyzers cannot read a `@typedef` within the same comment block as a function. No warning is shown, but the autocomplete features do not work.

Maybe it would be better to make it two comment blocs: one for the `@typedef` and another one for the function, with its `@example`s and `@return`. Or just describe the properties directly on the `@return`, without declaring a new type since it's not used elsewhere.

Also, both `error` and `address` are optional properties of this type, but are declared as mandatory.